marathon_lb.py

```diff
@@ -356,14 +359,27 @@ def config(apps, groups, bind_http_https, ssl_certs, templater):
             apps_with_http_appid_backend += [app.appId]
             cleanedUpAppId = re.sub(r'[^a-zA-Z0-9\-]', '_', app.appId)
 
-            http_appid_frontend_acl = templater \
-                .haproxy_http_frontend_appid_acl(app)
-            http_appid_frontends += http_appid_frontend_acl.format(
-                cleanedUpAppId=cleanedUpAppId,
-                hostname=app.hostname,
-                appId=app.appId,
-                backend=backend
-            )
+            if haproxy_map:
+                if 'map_http_frontend_appid_acl' not in duplicate_map:
+                    http_appid_frontend_acl = templater \
+                        .haproxy_map_http_frontend_appid_acl(app)
+                    http_appid_frontends += http_appid_frontend_acl.format(
+                        haproxy_dir="/".join(config_file.split("/")[:-1])
```

Comment:
Precompute this directory and store it in a val. I'd also prefer if you used `os.path` instead of rolling your own.